### PR TITLE
use `Base.ismutabletype`

### PR DESCRIPTION
**Pull request by mcabbott**
May fix a test failure on master.

### CODE REVIEW

File: src/lib/lib.jl [hunk 1]
```diff
@@ -1,5 +1,11 @@
 using Base: RefValue
 
+if VERSION > v"1.7.0-DEV.204"
+  using Base: ismutabletype
+else
+  ismutabletype(::Type{T}) where T = T.mutable
```
Comment (johnnychen94) [inline src/lib/lib.jl:6]:
```suggestion
  function ismutabletype(@nospecialize(t::Type))
    t = Base.unwrap_unionall(t)
    return isa(t, DataType) && t.mutable
  end
```

Comment (mcabbott) [reply src/lib/lib.jl:6]: Maybe better? It used to literally call `T.mutable` so that's all I was trying to preserve.

Comment (johnnychen94) [reply src/lib/lib.jl:6]: Just feel it better to keep consistent with the upstream implementation if we want to use the name `ismutabletype` 😆

Comment (mcabbott) [reply src/lib/lib.jl:6]: Yes am using that on new versions; CI informed me that it's a recent addition. 

I guess if there any difference in specialisation performance, we don't care as it's only used inside `@generated`, in fact would prefer less specialisation always.

Comment (mcabbott) [reply src/lib/lib.jl:6]: Let's call CI sufficiently happy

File: src/lib/lib.jl [hunk 2]
```diff
@@ -1,5 +1,14 @@
 using Base: RefValue
 
+if VERSION > v"1.7.0-DEV.204"
+  using Base: ismutabletype
+else
+  function ismutabletype(@nospecialize(t::Type))
```
Comment (ChrisRackauckas) [inline src/lib/lib.jl:6]: `@pure` this?

Comment (mcabbott) [reply src/lib/lib.jl:6]: It's copied straight from Base, I see this was discussed here: https://github.com/JuliaLang/julia/pull/39037/files#r550261314

**Pull request merged**